### PR TITLE
Tests for cumulative metrics queried with non-default granularities

### PR DESCRIPTION
**Pull request by courtneyholcomb**
Tests!!

### CODE REVIEW

File: tests_metricflow/integration/test_cases/itest_cumulative_metric.yaml [hunk 1]
```diff
@@ -412,3 +412,183 @@ integration_test:
...
+      , revenue_instance__ds__year
+      , revenue_mtd
+    FROM (
+      SELECT
+        revenue_instance__ds__quarter
+        , revenue_instance__ds__year
+        , FIRST_VALUE(revenue_mtd) OVER (
+          PARTITION BY revenue_instance__ds__quarter, revenue_instance__ds__year
+          ORDER BY metric_time__day
+          ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING
```
Comment (tlento) [inline tests_metricflow/integration/test_cases/itest_cumulative_metric.yaml:518]: Whoa, all of this works on all engines?

File: tests_metricflow/integration/test_cases/itest_cumulative_metric.yaml [hunk 2]
```diff
@@ -412,3 +412,183 @@ integration_test:
...
+  check_query: |
+    SELECT
+      revenue_instance__ds__quarter
+      , revenue_instance__ds__year
+      , revenue_mtd
+    FROM (
+      SELECT
+        revenue_instance__ds__quarter
+        , revenue_instance__ds__year
+        , FIRST_VALUE(revenue_mtd) OVER (
```
Comment (tlento) [inline tests_metricflow/integration/test_cases/itest_cumulative_metric.yaml:515]: Oh this is interesting. If it's month to date and people truncate to year that's super strange, right? It's just always month to date for January or whatever? I guess the answer is "don't do that" - but I do wonder if we should have a different default from FIRST_VALUE - I guess people can eventually override it to LAST_VALUE (which might be more sensible).

**Review by tlento — APPROVED** [review]
ted lasso woohoo

File: tests_metricflow/integration/test_cases/itest_cumulative_metric.yaml (hunk 1 above)
Comment (courtneyholcomb) [reply tests_metricflow/integration/test_cases/itest_cumulative_metric.yaml:518]: This SQL rendering logic already existed for conversion metrics, so it should!

Comment (courtneyholcomb) [reply tests_metricflow/integration/test_cases/itest_cumulative_metric.yaml:518]: And also SQL engine & check query tests pass, so yes.

**Pull request merged**